### PR TITLE
[E]ating menu stays open on last item

### LINKED ISSUE #23178: Eat from this stack until full
In the "Consume item" menu it would be nice to have a feature like Shift+F to eat/drink from a given stack of same food items until the stack is depleted or your character can no longer eat/drink anymore (excluding things that might have side effects like medicine and alcohol).

*Edit* I think it would make more sense to eat of the same item type instead of stack... >.>

### LINKED ISSUE #26879: Eat last food key.
**Is your feature request related to a problem? Please describe.**  
Well, you ate something, and still hungry and you have to scroll down through this again just to find that one loaf of bread you have laying on the ground? 
image
Isn't that frustrating? Sure, you can just press / to filter out the food you need, but that's almost as bad as scrolling through the list.

**Describe the solution you'd like**  
Why not add a simple 'Eat last eaten/consumed food' key? Which is obviously as similar as 'Craft last item' key!

**Describe alternatives you've considered**  
Honestly I have no other alternatives, other ideas are welcomed though!

**Additional context**  
Thank you again for reading and making the Favorites/Recent tab a thing, using it a lot! c:

### PR DESCRIPTION
**Pull request by cafeoh**
#### Summary
```SUMMARY: Interface "[E]ating menu now stays open on the last consume item"```
_This commit is dependent on ~~#29129~~ #29287 now I guess_

FabulousSmoothConure-mobile

#### Purpose of change
Fixes #26879
Fixes #23178

The ability to streamline the consumption of items and avoid repetitive menuing has been requested for quite some time, as continuously reopening the menu and scrolling down a potentially long list of consumables can be annoying and prone to errors.

#### Describe the solution
The net effect of this change is that the selection menu will now stay open after consuming an item, and keep the selection cursor where it was.

This commit defines a new activity (`ACT_EAT`), that will reopen the menu automatically once the item has been consumed, and reselect the previous entry.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -9290,6 +9297,15 @@ void game::eat( int pos )
             item_loc.remove_item();
         }
     }
+    g->refresh_all();
+
+    // Set up activity to reopen this menu next turn
+    if( u.has_activity( activity_id( "ACT_EAT" ) ) ) {
+        u.activity.targets.clear();
```
Comment (kevingranade) [inline src/game.cpp]: Need to make sure this doesn't have side effects. As we continue to increase the number of activities, clearing the whole list gets more and more risky.

**Comment by cafeoh** [comment]
This doesn't clear the whole list of activity (unless I'm using targets in an illegal way?), it should only clear the targets (vector of item_location) from the last list.

Currently (short of a bug) targets should never have more than one element here.

File: src/game.cpp (hunk 1 above)
Comment (cafeoh) [reply src/game.cpp]: This doesn't clear the whole list of activity, u.activity.targets is just a vector of item_location the activity carries.
Currently, short of a bug, targets should never have more than one element here.

**Comment by Meuka** [comment]
I must say, I'm really excited about this PR.

**Comment by kevingranade** [comment]
Seems to work ok, but I'm not clear on why the item favoriting is in here... Need to read the code a bit more.

**Comment by cafeoh** [comment]
> 
> Seems to work ok, but I'm not clear on why the item favoriting is in here... Need to read the code a bit more.

Yeah I built the changes on the *first* commit of the favorite system (was in another PR that I closed), because I changed parts of the menus. The first PR I marked it dependent to is closed, but #29287 builds on the first commit too.

I can rebase the favoriting change out if you want, or if you just take look at #29287 first, it'll be easier if/once this one gets merged.

**Comment by Rivet-the-Zombie** [comment]
What's the status of this PR?

**Comment by cafeoh** [comment]
I started remaking it from scratch. I'm not super happy with it, it's still pretty hacky and I don't have much time on my hands these days so can't give an ETA, but whenever I can dedicate an hour or two to it, I'll push a new commit out.

**Comment by cafeoh** [comment]
Alright it's good to go now.

Code isn't perfect, I decided to have some more specific code in inv_internal, and that's not amazing, but any other menu using `inv_internal` could also just use the same system (create an activity beforehand and let the menu push both the selected column and entry to `activity.values` for reuse).

Alternative would be a pretty length rework of UI, storing menu related entries in the player object (which was my second choice), or generally carrying values back up the stack, which I'm really not fond of.

Of course I'm open to suggestions if someone sees a cleaner way to do this.

**Review by ifreund — APPROVED** [review]
Code looks good to me!

**Pull request merged**